javascript/organizing_your_javascript_code/factory_functions_and_module_pattern.md

```diff
@@ -1,305 +1,250 @@
+We've introduced a new metric for a new user, a reputation, and in our returned object, added two new functions to control this reputation count. Sounds about right? But notice that we did not return the reputation count, *itself*. Instead, we used the power of closure to let it remain hidden, as a detail of implementation, and offered a function to check its value at any time and another one to increase its value by one. This is what we call a "private" variable.
 
-What would happen here if we tried to call `jimmie.die()`? What if we tried to manipulate the health: `jimmie.health -= 1000`? Well, those are things that we have NOT exposed publicly so we would get an error. This is a very good thing! Setting up objects like this makes it easier for us to use them because we've actually put some thought into how and when we are going to want to use the information. In this case, we have jimmie's health hiding as a private variable inside of the object which means we need to export a function if we want to manipulate it. In the long run, this will make our code _much_ easier to reason about because all of the logic is encapsulated in an appropriate place.
+Concerning factory functions, a private variable or function uses closures to create smaller, dedicated variables and functions within a factory function itself - things that we do not *need* to return in the object itself. This way we can create neater code, without polluting the returned object with unnecessary variables that we create while creating the object itself. Often, you do not need every single function within a factory to be returned with the object, or expose an internal variable. You can use them privately since the property of closures allows you to do so.
 
-#### Inheritance with factories
+In this case, we did not need control of `reputation` itself. To avoid foot guns, like accidentally setting the reputation to `-18000`, we simply expose the necessary details in the form of `getReputation` and `giveReputation`.
 
-In the constructors lesson, we looked fairly deeply into the concept of prototypes and inheritance, or giving our objects access to the methods and properties of another object. There are a few easy ways to accomplish this while using factories. Check this one out:
+### Prototypal inheritance with factories
+
+In the lesson with constructors, we looked deeply into the concept of prototype and inheritance, and how to give our objects access to the properties of another. With factory functions too, there are easy ways to do that. Take another hypothetical scenario into consideration. We need to extend the `User` factory into a `Player` factory that needs to control some more metrics - there are some ways to do that:
 
 ~~~javascript
-const Person = (name) => {
-  const sayName = () => console.log(`my name is ${name}`);
-  return {sayName};
-}
+function createPlayer (name, level) {
+  const { discordName, getReputation } = createUser(name);
```

Comment:
Since destructuring isn't mentioned anywhere in the previous lessons, I think we should also add the same comment here for clarity.

Reply:
Would a note on this, separately, be a better idea? Like the one with object shorthand.

Reply:
I would probably just explain it along with the object shorthand stuff. I feel like the concepts play off of each other.